include/FanFicArchive.py

```diff
@@ -0,0 +1,85 @@
+from bs4 import BeautifulSoup; from urllib import request; from re import match; import requests
+
+parser = "lxml"
+
+def fanfic_req(link): return BeautifulSoup(request.urlopen(link).read(), parser)
+def fanfic_check(id, name):
+    if isinstance(id, int): id = str(id)
+    else: assert isinstance(id, str), "Error for" + name + ": id not int or str"
+
+def ao3_rip(ao3_id): # downloads whole story in one go
+    fanfic_check(ao3_id, "ao3_rip")
+    text = fanfic_req('http://archiveofourown.org/works/'+ao3_id+'?view_adult=true&view_full_work=true')
+    tag_bar = text.find("dl", class_="stats").find_all("dd") # everything in the top bar
+    tag_list = {}
+    for i in tag_bar:
+        tag = [j.get_text() for j in i.find_all("a", class_="tag")]
+        if tag == []: tag = [i.get_text().lstrip('\n').lstrip(' ').rstrip(' ').rstrip('\n')]
+        tag_list[i.get("class")[0]] = tag
+    tag_list['author'] = [i.get_text() for i in text.find_all("a", rel="author")]
+    summary = text.find_all("div", class_="summary module", role="complementary") # just below top bar
+    chapter = {}
+    for i in text.find_all("div", class_="chapter", id=True): # separating each chapters
+        x = i.find("h3", class_="title").find("a").get("href").split("/")[-1]
+        chapter[i.get('id')] = [x, i]
+    return ao3_id, tag_list, summary, chapter
+
+def ff_rip(ff_id): # no image
+    fanfic_check(ff_id, "ff_rip")
+    text = fanfic_req("https://www.fanfiction.net/s/" +ff_id +"/1") # first chapter
```

Comment:
`ff_id` may be an `int` at this point. `fanfic_check` does not convert it to a string, because it cannot modify the `ff_id` variable from this scope.

Reply:
On second thoughts, consider replacing `fanfic_check(...)` with `ff_id = str(ff_id)`. If it is already a string, this will do nothing. If it is a number, it will convert it to a string. If it is something else, you'll get a 404 anyway.

Reply:
So I didn't do strict type checks on this one as it was python2 and not python3